cognitive/src/main/scala/com/microsoft/azure/synapse/ml/services/openai/GlobalParam.scala

```diff
@@ -0,0 +1,161 @@
+package com.microsoft.azure.synapse.ml.services.openai
+
+import com.microsoft.azure.synapse.ml.param.ServiceParam
+import com.microsoft.azure.synapse.ml.services.HasServiceParams
+import org.apache.spark.ml.param.Param
+import org.apache.spark.sql.Row
+
+import scala.collection.mutable
+
+sealed trait GlobalParamKey[T]
+case object OpenAIDeploymentNameKey extends GlobalParamKey[String]
+
+object GlobalParamObject {
+  private val ParamToKeyMap: mutable.Map[Any, GlobalParamKey[_]] = mutable.Map.empty
+  private val GlobalParams: mutable.Map[GlobalParamKey[Any], Any] = mutable.Map.empty
+
+  def setGlobalParamKey[T](key: GlobalParamKey[T], value: T): Unit = {
```

Comment:
you might want to make as many things private as you can here to defend against the dark arts

Reply:
Made this function private as I added explicit setters and getters, don't think I can make anything else private.